### ISSUE TITLE
Using `pivot` as a CTE name raises unparsable error

### ISSUE THREAD

**Issue by alanmcruickshank** [sqlfluff/sqlfluff#2165]
## Expected Behaviour

This completely valid snowflake query should parse fine:

```sql
with pivot as (select 1)
select * from pivot
```

## Observed Behaviour

`sqlfluff lint` raises:

```
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'with
                       | pivot as (select 1)\r\nselect * from ...'
```

...however if the query is changed to...

```sql
with pivoted as (select 1)
select * from pivoted
```

...then no issues is raised.

This leads to the conclusion that the classification of `pivot` as a keyword is being a little over-zealous at the moment.

## Dialect

Snowflake dialect.

## Version

Confirmed on version `0.7.0` & `0.9.0`. Running dbt templater but I don't think it's related.

## Configuration
```
[sqlfluff]
dialect = snowflake
templater = dbt
rules = L004,L010,L011,L014,L018,L023,L024,L026,L033,L035,L040,L045,L047,L048
exclude_rules = L009

[sqlfluff:rules:L010]
capitalisation_policy = lower

[sqlfluff:rules:L014]
capitalisation_policy = lower
```

**Comment by BorePlusPlus** [comment]
I think there may have been some issues introduced at the last improvement of Snowflake dialect. I am experiencing similar issues when using `account` as a name and it gets picked up as a keyword. See: https://github.com/sqlfluff/sqlfluff/issues/1927

On the other hand, I just saw that you say you have the issue with `0.7.0` as well. In my case downgrading to `0.7.1` (version prior to Snowflake dialect improvements) clears the issue. 

I am using jinja template in case that matters.
